arcgis-ios-sdk-samples/Content Display Logic/Controllers/OptionsTableViewController.swift

```diff
@@ -19,11 +19,13 @@ import UIKit
 class OptionsTableViewController: UITableViewController {
     
     private let labels: [String]
+    private let imageNames: [String]
     private var selectedIndex: Int
     private let onChange: (Int) -> Void
     
-    init(labels: [String], selectedIndex: Int, onChange: @escaping (Int) -> Void) {
+    init(labels: [String], imageNames: [String] = [], selectedIndex: Int, onChange: @escaping (Int) -> Void) {
```

Comment:
One of the issues I saw with `CustomContextSheet` was that it took a separate array for the titles, images, and highlight images. Instead, it should have taken an array of a type with a property for the title, the image, and the highlight image.

Could you please combine the title and image into a single type? That could either be a protocol or a concrete type:

```swift
struct Option {
    var title: String
    var image: UIImage?
}
```

Using `UIImage` for `image` makes sense because some options may already have `UIImage` objects that didn't come from named resources.

Reply:
I mean this is theoretically better but it puts an unnecessary burden on samples to wrap the titles in an `Option` type when there isn't an image. If you're set on it then I'd make another initializer.

Reply:
> it puts an unnecessary burden on samples to wrap the titles in an Option type

Mapping to another type isn't a burden. Consumers already have to map over their underlying type to get an array of titles, so updating the closure to return an `Option` rather than a `String` is rather trivial.

Another option would be to make `Option` a protocol:

```swift
protocol OptionProtocol {
    var title: String { get }
    var image: UIImage? { get }
}

extension OptionProtocol {
    // Convenience for options that do not have a corresponding image.
    var image: UIImage? { return nil }
}
```

And then make `OptionsTableViewController` generic. The advantage is that the underlying type can conform to `Option`, eliminating the need to do any wrapping/mapping. I see that as the best "option", pun intended.

Reply:
Some of the consumers don't have an underlying type, they just pass in an array of titles directly. Even if they all did, using a protocol would mean extending the SDK and I'm against that. Both options make the sample unnecessarily more complicated.

As far as passing in images rather than image names, that would be a good option if there were use case that necessitated it. For now it's cleaner to let the options controller handle image initialization.